### ISSUE TITLE
Remaining tabs are not resized when closing tabs

### ISSUE THREAD

**Issue by DavidZidar** [microsoft/terminal#9822]
### Windows Terminal version (or Windows build number)

1.8.1032.0

### Steps to reproduce

Open a bunch of tabs, close all but a few with the keyboard.

### Expected Behavior

I expect tabs to be restored to their original size when the space allows for it.

### Actual Behavior

Tabs remain compacted indefinitely. The only way to get the tabs back to their expected size seems to be to grab the mouse and hover over the tab bar and then move the mouse off the tab bar again.

**Comment by DavidZidar** [comment]
I understand the tabs remain fixed in size until the mouse leaves the tab bar like what was mentioned in #8030, but this makes no sense when closing tabs with the keyboard or even when just exiting shells while the mouse is nowhere near the tab bar.

**Comment by DHowett** [comment]
We suspect that this is an issue with MUX 2.5 :|

**Comment by DavidZidar** [comment]
Should I report this issue at microsoft/microsoft-ui-xaml or hold off until you are sure?

**Comment by DHowett** [comment]
Oh, sorry -- I'll move this issue report over there. I wasn't very clear on that. :)

**Comment by DavidZidar** [comment]
Thank you! And thanks for Windows Terminal, it's awesome! :)

**Comment by zadjii-msft** [comment]
I've moved this upstream to /dup https://github.com/microsoft/microsoft-ui-xaml/issues/4971. Thanks!

**Comment by beervoley** [comment]
close in https://github.com/microsoft/microsoft-ui-xaml/pull/6160
@zadjii-msft

**Comment by zadjii-msft** [comment]
@beervoley Are we _sure_ that https://github.com/microsoft/microsoft-ui-xaml/pull/6160 fixed this? I just whipped up [a Terminal build with 2.8](https://github.com/microsoft/terminal/compare/dev/migrie/mux-2.8-aug-2022) and this doesn't seem like it's fixed. 

![gh-9822-aug-2022](https://user-images.githubusercontent.com/18356694/182419669-803672d0-0387-4980-a8b2-97fff9d8ee46.gif)


(the numberbox at the start of the gif shows that I've definitely got 2.8, b/c #13495 is fixed)

As I close tabs, they still don't seem to resize. Maybe it's a side effect of how we're removing the tab items from the list?

https://github.com/microsoft/terminal/blob/7976e48195a569c87e37b7ca4861dfdcb7db0d7c/src/cascadia/TerminalApp/TabManagement.cpp#L532-L534

But that seems... pretty straightforward.

**Comment by StephenLPeters** [comment]
IDK why I looked at this, the email cought my eye, but this is also not correct: https://github.com/microsoft/microsoft-ui-xaml/blob/9670f769069535175490ee611e8a1ddf3fffd094/dev/TabView/TabView.cpp#L780  It means that if your mouse happens to be over tab item one in the tab strip when you use keyboard to close tab 7 they wont resize.  I don't think its the issue you are seeing though @zadjii-msft. @ranjeshj

**Comment by beervoley** [comment]
> @beervoley Are we _sure_ that [microsoft/microsoft-ui-xaml#6160](https://github.com/microsoft/microsoft-ui-xaml/pull/6160) fixed this? I just whipped up [a Terminal build with 2.8](https://github.com/microsoft/terminal/compare/dev/migrie/mux-2.8-aug-2022) and this doesn't seem like it's fixed.
> 
> ![gh-9822-aug-2022](https://user-images.githubusercontent.com/18356694/182419669-803672d0-0387-4980-a8b2-97fff9d8ee46.gif) [ ![gh-9822-aug-2022](https://user-images.githubusercontent.com/18356694/182419669-803672d0-0387-4980-a8b2-97fff9d8ee46.gif) ](https://user-images.githubusercontent.com/18356694/182419669-803672d0-0387-4980-a8b2-97fff9d8ee46.gif) [ ](https://user-images.githubusercontent.com/18356694/182419669-803672d0-0387-4980-a8b2-97fff9d8ee46.gif)
> 
> (the numberbox at the start of the gif shows that I've definitely got 2.8, b/c #13495 is fixed)
> 
> As I close tabs, they still don't seem to resize. Maybe it's a side effect of how we're removing the tab items from the list?
> 
> https://github.com/microsoft/terminal/blob/7976e48195a569c87e37b7ca4861dfdcb7db0d7c/src/cascadia/TerminalApp/TabManagement.cpp#L532-L534
> 
> But that seems... pretty straightforward.

yup, seeing the same
as @StephenLPeters outlined in a comment above - there's still a bug that if the cursor is over tabs and a tab is removed via keyboard - it still won't resize but that doesn't look what you're seeing.
![image](https://user-images.githubusercontent.com/21356912/182445424-fca5e382-c526-4bbf-b278-865af1e3be85.png)
not sure why this doesn't kick in: your pointer is clearly not over tabs so tabs width update should be triggered
@StephenLPeters any thoughts? going over it with debugger should point the issue right away

**Comment by StephenLPeters** [comment]
I didn't see anything wrong with my cursory review, but maybe has to do with xaml island stuff? maybe we don't get a pointer exited because the tab well goes to the edge of the island?

**Comment by beervoley** [comment]
@ranjeshj the fix for this particular problem (not when cursor is still over tabs and a tab is closed via keyboard) is to change line 782 to `UpdateTabWidths()` instead of `UpdateTabWidths(true, false)`. This will force TabView to use all the available width to size the tabs. I don't know why we need `fillAllAvailableSpace` bool in `UpdateTabWidths` at all. Passing `false` to `shouldUpdateTabWidths` will "achieve" the same result in case we don't need to resize tabs while user's cursor is still located over them.
So to summarize:
1) Update line 782 to `UpdateTabWidths()` to resize the tabs properly on keyboard close.
2) Keyboard close should be a distinct event from mouse click close because if the user's cursor is located on the tabs and a tab is closed via keyboard - tabview won't resize.

**Comment by zadjii-msft** [comment]
Upstream: https://github.com/microsoft/microsoft-ui-xaml/issues/8358

**Comment by mirumirumi** [comment]
I can still reproduce this in 2026. I primarily close tabs using a keyboard shortcut, while the pointer is nowhere near the tab strip.

  After closing enough tabs for more space to become available, the remaining tabs keep their narrow width indefinitely. The reliable way to restore
  the expected equal width is to move the pointer into the tab strip and then out again.

  This is distinct from the intentional deferral used when repeatedly clicking tab close buttons: there is no pointer-based close interaction in
  this scenario, so retaining the old width provides no UX benefit and is particularly disruptive in keyboard-heavy workflows.

  I also observe the same user-visible behavior in Notepad and File Explorer, which seems consistent with this being tracked upstream in microsoft/
  microsoft-ui-xaml#8358.

  Environment:
  - Windows Terminal: 1.24.11911.0
  - Windows 11: 26200.8894
